twins/service.go

```diff
@@ -293,12 +306,18 @@ func (ts *twinsService) saveState(msg *mainflux.Message, id string) error {
 	}
 
 	for _, rec := range recs {
-		if save := prepareState(&st, &tw, rec, msg); !save {
+		action := prepareState(&st, &tw, rec, msg)
+		switch action {
+		case 0:
```

Comment:
you have constants for `action` , use them

Reply:
Done.